system/include/emscripten/proxying.h

```diff
@@ -54,6 +54,17 @@ int emscripten_proxy_async(em_proxying_queue* q,
                            void (*func)(void*),
                            void* arg);
 
+// Enqueue `func` on the given queue and thread. Once it finishes executing, it
+// will proxy `callback` back to the current thread on the same queue. Returns 1
+// if the work was successfully enqueued and the target thread notified or 0
+// otherwise.
```

Comment:
Reading this, it seems like there are two enqueuings, is that right? We enqueue work on another thread, then it enques work back on us. (1) Are both async? (2) Does "Returns 1 if the work was successfully enqueued" refer to the first or the second enqueuing? It seems like there are two targets, one for each, so either could fail to enqueue, and the text seems a little ambiguous (around "target", and which enqueuing), but maybe I'm reading it wrong.

Reply:
Right, there are two enqueuings. Both are async, and since the callback is enqueued arbitrarily far in the future, we have no way of reporting an error if it fails to be enqueued for some reason. I'll update the text to be more precise.